Source/Data/HACompressedEntity.swift

```diff
@@ -0,0 +1,75 @@
+import Foundation
+
+public struct CompressedStatesUpdates: HADataDecodable {
+    public let add: [String: CompressedEntityState]?
+    public let remove: [String]?
+    public let change: [String: CompressedEntityDiff]?
+
+    public init(data: HAData) throws {
+        self.add = try? data.decode("a")
+        self.remove = try? data.decode("r")
+        self.change = try? data.decode("c")
+    }
+}
+
+public struct CompressedEntityState: HADataDecodable {
+    public let state: String
+    public let attributes: NSDictionary?
+    public let context: String?
+    public let lastChanged: Double?
+    public let lastUpdated: Double?
+
+    public init(data: HAData) throws {
+        self.state = try data.decode("s")
+        self.attributes = try? data.decode("a")
+        self.context = try? data.decode("c")
+        self.lastChanged = try? data.decode("lc")
+        self.lastUpdated = try? data.decode("lu")
+    }
+
+    public var lastChangedDate: Date? {
```

Comment:
see `HADecodeTransformable` - it already handles dates; please add support for Double-based dates as well and you won't need to do this